ludwig/automl/automl.py

```diff
@@ -0,0 +1,91 @@
+"""
+automl.py
+
+Driver script which:
+
+(1) Builds a base config by performing type inference and populating config
+    w/default combiner parameters, training paramers, and hyperopt search space
+(2) Tunes config based on resource constraints
+(3) Runs hyperparameter optimization experiment
+"""
+from logging import raiseExceptions
+from typing import Dict, Union
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+from ludwig.automl.base_config import create_default_config
+from ludwig.hyperopt.run import hyperopt
+
+OUTPUT_DIR = "."
+
+
+def model_select(default_configs):
+    """
+    Performs model selection based on dataset.
+    Note: Current implementation returns tabnet by default. This will be
+        improved in subsequent iterations
+    """
+    return default_configs['tabnet'], 'tabnet'
+
+
+def auto_train(
+    dataset: Union[str, pd.DataFrame, dd.core.DataFrame],
+    target: str,
+    time_limit_s: Union[int, float],
+    output_dir: str = OUTPUT_DIR
+):
+    """
+    Main auto train API that first builds configs for each model type
+    (e.g. concat, tabnet, transformer). Then selects model based on dataset
+    attributes. And finally runs a hyperparameter optimization experiment.
+
+    All batch and learning rate tuning is done @ training time.
+
+    # Inputs
+    :param dataset: (str) filepath to dataset.
+    :param target_name: (str) name of target feature
+    :param time_limit_s: (int, float) total time allocated to auto_train. acts
+                        as the stopping parameter
+
+    # Returns
+    :return: (str) path to best trained model
+    """
+
+    default_configs = create_default_config(dataset, target, time_limit_s)
```

Comment:
I think it could be useful to separate out the `create_default_config` in case the user wants to do something like: create default config, modify, then auto-train.

So perhaps we could do this:

```
def auto_train(..., default=None):
    if config is None:
        config = create_auto_config(dataset, target, time_limit_s)
```

This `create_auto_config` would then combine the default config creation with model selection. Maybe instead of returning the model name we can obtain this from the combiner name? That would also simplify the API for end users.

Reply:
@tgaddair this make sense. Just to clarify, we would expect that if the user wants to modify the config themselves, they would call `create_default_config` first before calling `auto_train` right?

Reply:
Yes, exactly.